mosaic/mosaic.go

```diff
@@ -0,0 +1,430 @@
+		{Char: '▄', Coverage: [4]bool{false, false, true, true}, CoverageMap: "  \n██"},   // Lower half block.
+		{Char: ' ', Coverage: [4]bool{false, false, false, false}, CoverageMap: "  \n  "}, // Space.
+		{Char: '█', Coverage: [4]bool{true, true, true, true}, CoverageMap: "██\n██"},     // Full block.
+	}
+	quarterBlocks = []block{
+		{Char: '▘', Coverage: [4]bool{true, false, false, false}, CoverageMap: "█ \n  "}, // Quadrant upper left.
+		{Char: '▝', Coverage: [4]bool{false, true, false, false}, CoverageMap: " █\n  "}, // Quadrant upper right.
+		{Char: '▖', Coverage: [4]bool{false, false, true, false}, CoverageMap: "  \n█ "}, // Quadrant lower left.
+		{Char: '▗', Coverage: [4]bool{false, false, false, true}, CoverageMap: "  \n █"}, // Quadrant lower right.
+		{Char: '▌', Coverage: [4]bool{true, false, true, false}, CoverageMap: "█ \n█ "},  // Left half block.
+		{Char: '▐', Coverage: [4]bool{false, true, false, true}, CoverageMap: " █\n █"},  // Right half block.
+		{Char: '▀', Coverage: [4]bool{true, true, false, false}, CoverageMap: "██\n  "},  // Upper half block (already added).
+		{Char: '▄', Coverage: [4]bool{false, false, true, true}, CoverageMap: "  \n██"},  // Lower half block (already added).
+	}
+	complexBlocks = []block{
+		{Char: '▙', Coverage: [4]bool{true, false, true, true}, CoverageMap: "█ \n██"},  // Quadrant upper left and lower half.
+		{Char: '▟', Coverage: [4]bool{false, true, true, true}, CoverageMap: " █\n██"},  // Quadrant upper right and lower half.
+		{Char: '▛', Coverage: [4]bool{true, true, true, false}, CoverageMap: "██\n█ "},  // Quadrant upper half and lower left.
+		{Char: '▜', Coverage: [4]bool{true, true, false, true}, CoverageMap: "██\n █"},  // Quadrant upper half and lower right.
+		{Char: '▚', Coverage: [4]bool{true, false, false, true}, CoverageMap: "█ \n █"}, // Quadrant upper left and lower right.
+		{Char: '▞', Coverage: [4]bool{false, true, true, false}, CoverageMap: " █\n█ "}, // Quadrant upper right and lower left.
+	}
+)
+
+// Block represents different Unicode block characters.
+type block struct {
+	Char        rune
+	Coverage    [4]bool // Which parts of the block are filled (true = filled).
+	CoverageMap string  // Visual representation of coverage for debugging.
+}
+
+// Symbol represents the symbol type to use when rendering the image.
+type Symbol uint8
+
+// Symbol types.
+const (
+	All Symbol = iota
+	Half
+	Quarter
+)
+
+// In many contexts, a default threshold level is often set to 0.5 (or 50%),
+// which means that values above this threshold are considered positive,
+// while those below are considered negative.
+// The value 128 represents the 0.5 of 0..255.
+const middleThresholdLevel = 128
+
+// Render mosaic with default values.
+func Render(img image.Image, width int, height int) string {
+	m := New().Width(width).Height(height)
+	return m.Render(img)
+}
+
+// Mosaic is a renderer and its options.
```

Comment:
```suggestion
// Mosaic represents a Unicode image renderer.
//
// Example:
//  ```go
//  art := mosaic.New().Width(100). // Limit to 100 cells
//      Scale(mosaic.Fit).                    // Fit to width
//      Render()
//  ```
```